File: frontend/src/types/session.ts
/**
 * TypeScript types for sessions
 */

import { AnnotatedEvent } from './annotation'

export interface Session {
  id: string
  session_id: string
  dataset: string
  user_id?: string
  start_time: string
  end_time: string
  num_events: number
  has_annotations: boolean
}

export interface SessionDetail extends Session {
  annotated_events: AnnotatedEvent[]
}

export interface SessionListResponse {
  sessions: Session[]
  total: number
  skip: number
  limit: number
}

export interface SessionStatistics {
  total_sessions: number
  total_events: number
  avg_events_per_session: number
  label_distribution: Record<string, number>
  dataset_breakdown: Record<string, number>
}

File: backend/app/schemas/session.py
"""
Pydantic schemas for session data
"""

from pydantic import BaseModel
from typing import List, Optional
from app.schemas.annotation import AnnotatedEvent


class SessionResponse(BaseModel):
    """Response containing session details"""
    session_id: str
    dataset: str
    user_id: Optional[str] = None
    start_time: str
    end_time: str
    num_events: int
    annotated_events: List[AnnotatedEvent]


class SessionSummary(BaseModel):
    """Summary of a session for list views"""
    session_id: str
    dataset: str
    num_events: int
    start_time: str
    has_annotations: bool


class SessionListResponse(BaseModel):
    """Response for paginated session list"""
    sessions: List[SessionSummary]
    total: int
    skip: int
    limit: int

